osi_sensordata.proto

```diff
@@ -60,143 +68,137 @@ message SensorData
     //
-    // \par References: 
+    // \par References:
     // [1] DIN ISO 8855:2013-11
     //
     // \note This field is usually static during the simulation.
-    // \note The origin of vehicle's coordinate system in world frame is 
+    // \note The origin of vehicle's coordinate system in world frame is
     // ( \c Vehicle::base.position + Inverse_Rotation_yaw_pitch_roll(\c Vehicle::base.orientation) * \c Vehicle::bbcenter_to_rear) .
-    // The orientation of the vehicle's coordinate system is equal to the orientation 
+    // The orientation of the vehicle's coordinate system is equal to the orientation
     // of the vehicle's bounding box \c Vehicle::base.orientation.
     //
-    optional MountingPosition mounting_position = 6;
+    optional MountingPosition mounting_position = 4;
 
     // The root mean squared error of the mounting position.
     //
-    optional MountingPosition mounting_position_rmse = 7;
+    optional MountingPosition mounting_position_rmse = 5;
+
+    // Sensor view w.r.t. the sensor coordinate system
+    //
+    // This provides a copy of the \c SensorView data received by the sensor
+    // for reference purposes.
+    //
+    optional SensorView sensor_view = 6;
 
     // The timestamp of the last real-world measurement (e.g. GT input) that
     // this set of sensor data takes into account.  This in effect is the last
     // time instance of reality the measurements correspond to.  See field
     // \c SensorData::timestamp for a detailed discussion.  This value is also
     // the upper bound to the \c DetectedObjectHeader::measurement_time and the
     // feature data \c SensorDetectionHeader::measurement_time fields.
-    // 
-    optional Timestamp last_measurement_time = 8;
+    //
+    optional Timestamp last_measurement_time = 7;
 
     // General information about the \c DetectedObject .
-    // 
-    optional DetectedEntityHeader object_header = 100;
-    
+    //
+    optional DetectedEntityHeader object_header = 8;
+
     // The list of objects detected by the sensor as perceived by the sensor.
     //
     repeated DetectedObject object = 9;
-    
+
     // General information about the \c DetectedTrafficSign .
-    // 
-    optional DetectedEntityHeader traffic_sign_header = 101;
+    //
+    optional DetectedEntityHeader traffic_sign_header = 10;
 
     // The list of traffic signs detected by the sensor
     //
-    repeated DetectedTrafficSign traffic_sign = 10;
-        
+    repeated DetectedTrafficSign traffic_sign = 11;
+
     // General information about the \c DetectedTrafficLight .
-    // 
-    optional DetectedEntityHeader traffic_light_header = 102;
+    //
+    optional DetectedEntityHeader traffic_light_header = 12;
 
     // The list of traffic lights detected by the sensor
-    // 
-    repeated DetectedTrafficLight traffic_light = 11;
+    //
+    repeated DetectedTrafficLight traffic_light = 13;
```

Comment:
See above, I tried to keep all naming stuff outside of this PR.

Reply:
As an aside, since we discussed this on the phone: Given the code that protoc generates for repeated fields, using the singular might be slightly less misleading than the plural (see discussion in https://stackoverflow.com/questions/4121175/protobuf-naming-conventions point 4):

For

    repeated DetectedTrafficLight traffic_light = 13;

the generated methods might be

    ::traffic_light(int index), ::traffic_light(), ::add_traffic_light()

this is slightly misleading due to the ::traffic_light(int index) vs. ::traffic_light() accessors (get one or all traffic_lights), however the plural version is also at least as misleading, where every method is called traffic_lights:

    ::traffic_lights(int index), ::traffic_lights(), ::add_traffic_lights()

So basically protobufs suck...

Reply:
We stick to the plural now. At least you can read it in a better way in the .proto file

Reply:
I'd still leave this PR as is, since the plural/singular stuff should be dealt with in a global commit I think (it is mixed up enough as it currently stands).

Reply:
@pmai I changed this (plural->singular) in the review PR. I can make the merge manually.